### ISSUE TITLE
Erroneous invalid-sequence-index on typing List

### ISSUE THREAD

**Issue by jmcs** [PyCQA/pylint#867]
Having something like:

    from typing import List
 
    def my_list() -> List[str]:
        return ['my string']

Raises an `invalid-sequence-index` because `typing.List` inherits from `list`, but this is obviously a false positive.

**Comment by Chiel92** [comment]
Same problem here.

**Comment by MichaelZapataScality** [comment]
I encounter the same issue.

**Comment by PCManticore** [comment]
Thanks for reporting this. I'll look into supporting typing's classes as soon as I'll get some time.

**Comment by PCManticore** [comment]
This fixes this particular occurence. The problem was one of the generic base classes of typing's List, which was contributing a custom ``__getitem__`` to the List class. The check we have for invalid-sequence-index looks to see if the owner of the ``__getitem__`` is one of the known sequences and if so, it checks the type of the index, which is expected to be an int. In this case, the owner of the ``__getitem__`` wasn't known, thus we had to ignore it in this particular case.

**Comment by tuukkamustonen** [comment]
@PCManticore With latest pylint 1.6.4, following still hits `invalid-sequence-index`:

```python
def my_list() -> Tuple[str, int]:
    return 'str', 1
```

Is that maybe another special case that needs special attention?

**Comment by PCManticore** [comment]
@tuukkamustonen I cannot reproduce this using 1.64, can you show me what exactly does this outputs for you?

**Comment by tuukkamustonen** [comment]
@PCManticore `foobar.py`:

    from typing import Tuple

    def my_list() -> Tuple[str, int]:
        return 'str', 1

`pylint --reports=no foobar.py`:

    ************* Module foobar
    foobar.py:4: [E1126(invalid-sequence-index), my_list] Sequence index is not an int, slice, or instance with __index__

**Comment by PCManticore** [comment]
This is weird because I am getting
```
************* Module a
C:  1, 0: Missing module docstring (missing-docstring)
C:  3, 0: Missing function docstring (missing-docstring)
```
Are you using astroid 1.4.8, no?

**Comment by tuukkamustonen** [comment]
@PCManticore Yeah, `astroid==1.4.8`:

```
$ mkvirtualenv --python /usr/bin/python3 py3
...
$ python --version
Python 3.4.3
$ pip install pylint mypy-lang
...
$ pip list
astroid (1.4.8)
isort (4.2.5)
lazy-object-proxy (1.2.2)
mccabe (0.5.2)
mypy-lang (0.4.6)
pip (1.5.4)
pylint (1.6.4)
setuptools (2.2)
six (1.10.0)
wrapt (1.10.8)
$ pylint --reports=no foobar.py
************* Module foobar
foobar.py:4: [E1126(invalid-sequence-index), my_list] Sequence index is not an int, slice, or instance with __index__
```

How is it that even the output looks different?

**Comment by PCManticore** [comment]
I'm using the default message template, while you probably have a custom one. The only difference that I now see is that you are using Python 3.4, while I am using Python 3.5, apart of this, everything seems the same. I suspect there are some differences in typing.py in these two interpreters, which leads to this error you are having. Let's open another issue and move the discussion there.

**Comment by jaysonsantos** [comment]
It will also break if you do this:
```python
def test():
    """"Test function."""
    data = []  # type List[int]
    for i in (1, 2, 2, 3, 5, 1, 2, 3):
        if i not in data:
            data.append(i)
````
using pylint pylint==1.6.5 and python 3.5
